src/util/StringUtils.h

```diff
@@ -142,6 +142,20 @@ std::string getLowercaseUtf8(std::string_view s) {
   return result;
 }
 
+// Get the uppercase value. For details see `getLowercaseUtf8` above
+inline std::string getUppercaseUtf8(std::string_view s) {
+  std::string result;
+  icu::StringByteSink<std::string> sink(&result);
+  UErrorCode err = U_ZERO_ERROR;
+  icu::CaseMap::utf8ToUpper(
+      "", 0, icu::StringPiece{s.data(), static_cast<int32_t>(s.size())}, sink,
+      nullptr, err);
+  if (U_FAILURE(err)) {
+    throw std::runtime_error(u_errorName(err));
+  }
+  return result;
+}
+
```

Comment:
Given the almost identical code, how about a helper function `utf8StringTransform`, which takes a `std::string_view` and the transformation function as arguments?

Also, it looks like it would be more consistent to name the two functions based on this helper function `utf8StringToLower` (instead of `getLowercaseUtf8`) and `utf8StringToUpper` (instead of `getUppercaseUtf8`).